src/trunk.c

```diff
@@ -8964,39 +8965,52 @@ trunk_config_init(trunk_config        *trunk_cfg,
    // Inline what we would get from trunk_pivot_size(trunk_handle *).
    trunk_pivot_size = data_cfg->max_key_size + sizeof(trunk_pivot_data);
 
-   // Setting hard limit and over overprovisioning
+   // Setting hard limit and check configuration for over-provisioning
    trunk_cfg->max_pivot_keys = trunk_cfg->fanout + TRUNK_EXTRA_PIVOT_KEYS;
    uint64 header_bytes       = sizeof(trunk_hdr);
 
    uint64 pivot_bytes = (trunk_cfg->max_pivot_keys
                          * (data_cfg->max_key_size + sizeof(trunk_pivot_data)));
    uint64 branch_bytes =
       trunk_cfg->max_branches_per_node * sizeof(trunk_branch);
-   uint64 trunk_node_min_size = header_bytes + pivot_bytes + branch_bytes;
-   uint64 available_pivot_bytes =
-      cache_config_page_size(cache_cfg) - header_bytes - branch_bytes;
+   uint64 trunk_node_min_size   = header_bytes + pivot_bytes + branch_bytes;
+   uint64 page_size             = trunk_page_size(trunk_cfg);
```

Comment:
This redirects to the `cache_cfg`, right? It feels better to use the one from the `cache_cfg`, rather than creating an implicit ordering requirement.

Reply:
I see that there was one already below, but I still think this is better as `cache_page_size(cache_cfg)`.

Reply:
I'm not exactly sure what you are asking. In the old code, like below, there were two instances of this call to `cache_config_page_size()` which I tried to factor out into a local variable `page_size`.

```
8979    uint64 available_pivot_bytes =
8980       cache_config_page_size(cache_cfg) - header_bytes - branch_bytes;
[...]
8985    platform_assert(trunk_node_min_size < cache_config_page_size(cache_cfg),
```

Are you suggesting that I should retain the call to `cache_config_page_size()` and not call `trunk_page_size()` [ even though it's internally calling `cache_config_page_size()`?

I'm assuming that it's ok to refactor it out to a local variable `page_size`.

Reply:
Done. Revert back to use of `cache_config_page_size(cache_cfg)`